include/Graphs/PAG.h

```diff
@@ -561,6 +563,25 @@ class PAG : public GenericGraph<PAGNode,PAGEdge>
     {
         return mem->getSymId();
     }
+    /// Get PAG Edge according to LLVM value
+    /// return nullptr if an edge cannot be found
+    inline PAGEdgeSet getValueEdge(const Value *V)
```

Comment:
This should return a const reference?

Reply:
Given that `PAGEdgeSet` is `Set<const PAGEdge*>` and the Set is moved anyway, I see no advantage of a const reference. However, the comment is wrong.

Reply:
`Reference` can avoid C++ copy construction.
`const` will make the code more readable that the set will not be modified anymore.

Reply:
I don't get it. See this minimal example:
```
#include <set>

std::set<int> my_function() {
	std::set<int> my_set;
	for (int i = 0; i < 5; ++i) {
		my_set.emplace(i);
	}
	return my_set;
}

int main() {
	std::set<int> main_set = my_function();
	const std::set<int> onother_set = my_function();
}
```
my_function seems to copy the set. However, the compiler optimizes it into a move. So there is only one move.
This is the resulting IR of my_function (where my_function takes a set pointer):
```
define dso_local void @_Z11my_functionv(%"class.std::set"* noalias sret(%"class.std::set") align 8 %agg.result) #0 personality i8* bitcast (i32 (...)* @__gxx_personality_v0 to i8*) {
```

Returning a reference alone would be wrong since it is a reference to a local variable, which does not exist anymore after the function ends.
Returning a const reference alows the compiler to move instead of returning a real reference, too. However, the caller cannot modify the set anymore. Why should the function force this?

Return just the set allows the caller to make it const or to modify it.

Reply:
I usually do not trust compiler optimizations:). The optimization usually depends on different compilers and options used. In your case, it is not a local variable but `valueToEdgeMap ` is a class member. I did not see anything wrong that you return a reference.

For `const`, the caller (invoking `getValueEdge `) should only read but not modify the set as your name suggests.

Reply:
Sorry, have confused this function with `fromLLVMValue`. Of course a const reference is meaningful here.
A problem arises, if an empty set should be returned. Therefore, I have put that dummy value at `valueToEdgeMap[nullptr]`.